src/Sparrow/Json/BlittableJsonReaderObject.cs

```diff
@@ -27,10 +27,16 @@ public unsafe class BlittableJsonReaderObject : BlittableJsonReaderBase, IDispos
 
         public override string ToString()
         {
+            return GetJsonReader().ReadToEnd();
+        }
+
+        public TextReader GetJsonReader()
+        {
             var memoryStream = new MemoryStream();
             _context.Write(memoryStream, this);
             memoryStream.Position = 0;
-            return new StreamReader(memoryStream).ReadToEnd();
+
+            return new StreamReader(memoryStream);
```

Comment:
This allocates 1KB of memory on call. Since we are probably calling this per indexed result, and we can have a lot of those, let us see if we can pass the memory stream and the stream reader externally, so we can reuse them per indexing batch, rather than allocate once per doc